### DISCUSSION TITLE
Mobx destructuring (React)

### DISCUSSION THREAD

**Discussion by lgibso34** [mobxjs/mobx#2916] [Q&A]
I know that JS destructuring is not really allowed in mobx due to loss of reactivity, but is there any reason I can't destructure actions or functions (autoActions)?

Those aren't reactive so I feel that I am able to pull them out since they should be the same every render.

I tried searching for an existing discussion but couldnt find anything.

So, what are the rules for destructuring?

Thank you.

**Comment by urugator** [comment]
> JS destructuring is not really allowed in mobx due to loss of reactivity

Can you post an example of what do you mean by that?
